src/git-town-prepend

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+source "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/helpers/helpers.sh"
+
+
+function ensure_has_target_branch {
+  if [ -z "$target_branch_name" ]; then
+    echo_error_header
+    echo_error "No branch name provided."
+    exit_with_error newline
+  fi
+}
+
+
+function preconditions {
+  target_branch_name=$1
+  ensure_has_target_branch
+
+  child_branch_name=$INITIAL_BRANCH_NAME
+  ensure_is_feature_branch "$INITIAL_BRANCH_NAME" "Only feature branches can have parent branches."
+
+  parent_branch_name=$(parent_branch "$child_branch_name")
+  if [ "$parent_branch_name" != "$MAIN_BRANCH_NAME" ]; then
+    ensure_knows_parent_branches "$parent_branch_name"
+  fi
+
+  if [ "$(has_remote_url)" = true ]; then
+    fetch
+  fi
+
+  ensure_does_not_have_branch "$target_branch_name"
+
+  export RUN_IN_GIT_ROOT=true
+  export STASH_OPEN_CHANGES=true
+}
+
+
+function steps {
+  # sync the parent branch
+  for branch_name in $(ancestor_branches "$parent_branch_name"); do
+    sync_branch_steps "$branch_name"
+  done
+  sync_branch_steps "$parent_branch_name"
+
+  echo "create_and_checkout_feature_branch $target_branch_name $parent_branch_name"
+  echo_if_all_true "create_tracking_branch $target_branch_name" "$HAS_REMOTE" "$(hack_should_push)"
+
+  echo "delete_all_ancestor_entries"
+
+  # make the new branch the parent of the existing branch
+  echo delete_parent_entry "$child_branch_name"
```

Comment:
I don't think this line is needed as the next line sets the value and thus would overwrite any existing value.